wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -342,11 +342,11 @@ abstract class Wallet
     for {
       allUnspent <- spendingInfoDAO.findAllUnspent()
     } yield {
-      val confirmedUtxos = allUnspent.filter { utxo =>
+      val unconfirmed = allUnspent.filter { utxo =>
         HDAccount.isSameAccount(utxo.privKeyPath.path, account) &&
-        utxo.state == PendingConfirmationsReceived || utxo.state == BroadcastReceived
```

Comment:
Very subtle logical error. This filter will evaluate to true if 

1. It's the same account and the utxo state is `PendingConfirmationsReceived`
2. if `utxo.state == BroadcastReceived`.

That means regardless of account, the balance will get credited if the utxo state is `BroadcastReceived`. 

What we meant to check is if it (`utxo.state` is EITHER `PendingConfirmationsReceived` OR `BroadcastReceived`) AND the account is the same